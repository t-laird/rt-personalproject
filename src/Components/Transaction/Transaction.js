import React, { Component } from 'react';
import './Transaction.css';
import getKeyFromLS from '../../helpers/getKeyFromLS';
import { connect } from 'react-redux';


class Transaction extends Component {
  constructor() {
    super();
    this.state = {
      points: '',
      recipient: '',
      validInput: true
    };
  }

  handleInput = (e) => {
    const {value, name} = e.target;

    this.setState({[name]: value});
  }

  pointStatus() {
    const { points } = this.state;
    const parsePoints = parseInt(points);

    if (
      (isNaN(parsePoints) || 
      parsePoints > 100) &&
      points.length
    ) {
      return <span className="point-error">X</span>
    } else {
      return null;
    }
  }

  getRemainingPoints() {
    const { UserTransactions } = this.props;
    if (UserTransactions.length) {
      const recentTransactions = UserTransactions[UserTransactions.length - 1];
      const spentPoints = recentTransactions.sent.reduce( (total, transaction) => {
        total += transaction.point_value;
        return total;
      }, 0);
      const weeklyPoints = this.props.Group.weekly_points;
      return <span>remaining points: {weeklyPoints-spentPoints}</span>;
    } 
    return <span>login to send points!</span>;
  }

  async handleSubmit() {
    const send_id = 123;
    const receive_id = parseInt(this.state.recipient);
    const group_id = 123;
    const point_value = parseInt(this.state.points);

    const submitEvent = await fetch('http://localhost:3000/api/v1/eventtracking/new', {
      method: 'POST',
      headers: {
        'CONTENT-TYPE': 'application/json',
        'x-token': getKeyFromLS()
      },
      body: JSON.stringify({send_id, receive_id, group_id, point_value})
    });

    const submitResponse = await submitEvent.json();
    console.log(submitResponse);
  }

  recipientOptions() {
    const filterSelf = this.props.UserList.filter( user => user.name !== this.props.User.name);
    const userOptions = filterSelf.map( (user, index) => {
      return <option value={user.name} key={`userOption${index}`}>{user.name}</option>
    });

    return userOptions;
  }

  render() {
    return (
      <div className="Transaction">
        <div className="remaining-points">
          {this.getRemainingPoints()}
        </div>
        <div className="points">
          <input 
            type="text" 
            name="points" 
            placeholder="points" 
            value={this.state.points} 
            className="points"
            onChange={(e) => {this.handleInput(e)}} />
          {this.pointStatus()}
        </div>
        <select name="cars">
          {this.recipientOptions()}
        </select> 
        <button onClick={()=> {this.handleSubmit()}}>SEND!</button>
      </div>
    );
  }
}


const mapStateToProps = ( store ) => ({
  UserTransactions: store.UserTransactions,
  UserList: store.UserList,
  User: store.User,
  Group: store.Group
});


export default connect(mapStateToProps, null)(Transaction);