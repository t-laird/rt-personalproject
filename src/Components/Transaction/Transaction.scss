@import '../../Variables.scss';

.Transaction {
  position: relative;
  input {
    background-color: $red-light;
    color: $red;
    font-size: 34px;
    text-align: center;
    &::placeholder{
      color: #fff;
      font-size: 24px;
      position: relative;
      bottom: 2px;
    }
  }

  .current-points {
    background: linear-gradient(to right, rgba(227,245,253,1) 0%,rgba(241,250,254,1) 20%,rgba(241,250,254,1) 80%,rgba(227,245,253,1) 100%);
    border-bottom: 7px solid #b9e5fb;
    width: 770px;
    margin: 0 auto 30px;
    padding: 2px 25px 8px;
  }

  .current-span {
    background-color: $blue;
    padding: 10px 17px;
    font-family: "semplicitapro", sans-serif;
    font-weight: 900;
    font-size: 24px;
    margin-left: 5px
  }

  h3 {
    margin: 0 0 13px 20px
  }

  button {
    background-color: #fe5630;
    color: white;
    cursor: pointer;
    padding: 9px 22px 8px;
    font-size: 24px;
    margin-left: 15px;
    transition: background-color .15s ease-in-out;

    &:hover {
      background-color: darken($red, 8%);
    }
  }
  
  hr {
    margin: 23px 0 20px;
    opacity: .2;
  }

  input,
  button,
  select {
    outline: none;
    border: none;
  }

  select {
    font-size: 24px;
    background-color: #ffddd6;
    margin-left: 15px;
    width: 268px;
    border-radius: 0;
    background: url(assets/selectors.svg) no-repeat 96% $red-light;
    background-size: 24px;
    -webkit-appearance: none;
  }

  .award {
    display: flex;

    h3 {
      margin-top: 13px;
    }
  }

  .points-to {
    margin-left: 0;
  }

  .points-block {
    position: relative;
    background: linear-gradient(to right, rgba(227,245,253,1) 0%,rgba(241,250,254,1) 20%,rgba(241,250,254,1) 80%,rgba(227,245,253,1) 100%);
    border-top: 7px solid #b9e5fb;
    border-bottom: 7px solid #b9e5fb;
    padding: 20px 25px;
    width: 770px;
    margin: auto;


    h2 {
      margin-bottom: 13px;
    }

    input {
      width: 80px;
      margin: 0 15px;
    }
    .point-error {
      position: absolute;
      top: 25px;
      right: 10px;
      color: red;
    }
  }
}