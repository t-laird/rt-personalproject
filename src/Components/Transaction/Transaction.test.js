/* eslint-disable max-len */

import { mapStateToProps, Transaction } from './Transaction';
import { shallow } from 'enzyme';
import React from 'react';

describe('Transaction component tests', () => {
  let mockProps;
  let renderedTransaction;

  beforeEach(() => {
    mockProps = {
      UserTransactions: [{sent: [], received: []}],
      UserList: [
        {name: 'Ann', user_id: 1}, 
        {name: 'Jen', user_id: 2}, 
        {name: 'Tim', user_id: 3}, 
        {name: 'Alan', user_id: 4}
      ],
      User: {
        authrocket_id: "usr_0vXXFrAmpjSWXnr7n2x2m8",
        created_date: "2017-11-18T18:34:30.017Z",
        email: "t6r6l5@gmail.com",
        group_id: '19vns',
        user_id: 20,
        name: "Thomas Laird"
      },
      Group: {
        administrator_id: 957,
        created_date: "2017-10-18T18:34:30.017Z",
        group_id: 15,
        group_name: null,
        group_passphrase: "19vns",
        weekly_points: 1000
      }
    };

    renderedTransaction = shallow(<Transaction {...mockProps} />);
  });

  it('should match the snapshot', () => {
    expect(renderedTransaction).toMatchSnapshot();
  });

  it('should update the state of recipient field on input', () => {
    const mockEvent = { target: {name: 'recipient', value: 'abc'}};

    renderedTransaction.instance().handleInput(mockEvent);

    expect(renderedTransaction.state('recipient')).toEqual('abc');
  });

  it('should update the state of points field on input', () => {
    const mockEvent = { target: {name: 'points', value: '100'}};

    renderedTransaction.instance().handleInput(mockEvent);

    expect(renderedTransaction.state('points')).toEqual('100');
  });
  
  it('should update the state of note field on input', () => {
    const mockEvent = { target: {name: 'note', value: 'thanks'}};

    renderedTransaction.instance().handleInput(mockEvent);

    expect(renderedTransaction.state('note')).toEqual('thanks');
  });

  it('should set the transactionMessage to appropriately when user input is not found', () => {
    const expectedMessage = "We can't find your receiving user.  You should try again!";

    renderedTransaction.setState({
      recipient: 'sdkfdfsd'
    });

    renderedTransaction.instance().handleSubmit();

    expect(renderedTransaction.state('transactionMessage')).toEqual(expectedMessage);
  });

  it('should set the transactionMessage to a success message with valid recipient and input', async () => {
    global.localStorage = {
      getItem: () => null
    };

    window.fetch = jest.fn().mockImplementation( () =>
      Promise.resolve({
        json: () => Promise.resolve({
          status: 'success'

        })
      })
    );

    renderedTransaction.setState({
      recipient: 'Alan',
      points: '10'
    });

    await renderedTransaction.instance().handleSubmit();
    renderedTransaction.update();

    const expectedMessage = 'Successfully sent 10 points to Alan.';

    expect(renderedTransaction.state('transactionMessage')).toEqual(expectedMessage);
  });

  it('should populate the suggestions to display on recipient input', () => {
    const mockEvent = {target: {value: 'Al'}};

    renderedTransaction.instance().populateSuggestions(mockEvent);

    const expectedSuggestions = [
      "Alan"
    ];

    expect(renderedTransaction.state('suggestions')).toEqual(expectedSuggestions);
  });

  it('should render a NO USERS FOUND message if suggestions array is empty on input', async () => {
    renderedTransaction.setState({
      recipient: 'aslkfjasdf'
    });

    const suggestionsResult = renderedTransaction.instance().generateSuggestions();

    expect(suggestionsResult.props.children).toEqual('no users found');
  });
});

describe('mapStateToProps tests', () => {
  it('should get user and group info from store', () => {
    const mockStore = {
      User: {
        authrocket_id: "usr_0vXXFrAmpjSWXnr7n2x2m8",
        created_date: "2017-11-18T18:34:30.017Z",
        email: "t6r6l5@gmail.com",
        group_id: '19vns',
        user_id: 20,
        name: "Thomas Laird"
      },
      Group: {
        administrator_id: 957,
        created_date: "2017-10-18T18:34:30.017Z",
        group_id: 15,
        group_name: null,
        group_passphrase: "19vns",
        weekly_points: 1000
      }
    };

    const result = mapStateToProps(mockStore);
    expect(result.User).toEqual(mockStore.User);
    expect(result.Group).toEqual(mockStore.Group);
  });

  it('should get user transactions and userlist from store', () => {
    const mockStore = {
      UserTransactions: [{sent: [], received: []}],
      UserList: [{name: 'Ann', user_id: 1}, {name: 'Jen', user_id: 2}]
    };

    const result = mapStateToProps(mockStore);
    expect(result.UserTransactions).toEqual(mockStore.UserTransactions);
    expect(result.UserList).toEqual(mockStore.UserList);
  });
});