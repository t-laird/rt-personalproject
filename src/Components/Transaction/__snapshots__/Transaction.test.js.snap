// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Transaction
    Group={
        Object {
            "administrator_id": 957,
            "created_date": "2017-10-18T18:34:30.017Z",
            "group_id": 15,
            "group_name": null,
            "group_passphrase": "19vns",
            "weekly_points": 1000,
          }
    }
    User={
        Object {
            "authrocket_id": "usr_0vXXFrAmpjSWXnr7n2x2m8",
            "created_date": "2017-11-18T18:34:30.017Z",
            "email": "t6r6l5@gmail.com",
            "group_id": "19vns",
            "name": "Thomas Laird",
            "user_id": 20,
          }
    }
    UserList={
        Array [
            Object {
              "name": "Ann",
              "user_id": 1,
            },
            Object {
              "name": "Jen",
              "user_id": 2,
            },
            Object {
              "name": "Tim",
              "user_id": 3,
            },
            Object {
              "name": "Alan",
              "user_id": 4,
            },
          ]
    }
    UserTransactions={
        Array [
            Object {
              "received": Array [],
              "sent": Array [],
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="current-points"
>
          <div
                    className="remaining-points"
          >
                    <span
                              className="points"
                    >
                              1000
                    </span>
                    <h6>
                              REMAINING
                    </h6>
          </div>
          <div
                    className="divider"
          >
                    <h4>
                              SNAPS
                    </h4>
                    <span>
                              THIS WEEK
                    </span>
          </div>
          <div
                    className="received-points"
          >
                    <span
                              className="points"
                    />
                    <h6>
                              RECEIVED
                    </h6>
          </div>
</div>,
        <div
          className="award-header"
>
          <h4>
                    AWARD
          </h4>
</div>,
        <div
          className="points-block"
>
          <div
                    className="award"
          >
                    <div
                              className="send-input"
                    >
                              <h5
                                        className="send"
                              >
                                        Send
                              </h5>
                              <input
                                        name="points"
                                        onChange={[Function]}
                                        placeholder="QTY"
                                        type="text"
                                        value=""
                              />
                    </div>
                    <div
                              className="recipient-input"
                    >
                              <h5
                                        className="snaps"
                              >
                                        snaps to:
                              </h5>
                              <input
                                        name="recipient"
                                        onChange={[Function]}
                                        onKeyDown={[Function]}
                                        placeholder="find a teammate"
                                        type="text"
                                        value=""
                              />
                              <ul>
                                        
                              </ul>
                    </div>
          </div>
          <textarea
                    className="note-input"
                    name="note"
                    onChange={[Function]}
                    placeholder="add a message"
                    type="text"
                    value=""
          />
          <button
                    onClick={[Function]}
          >
                    SEND
          </button>
          <div
                    className="transaction-message"
          >
                    <h3 />
          </div>
</div>,
      ],
      "className": "Transaction",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="remaining-points"
>
              <span
                            className="points"
              >
                            1000
              </span>
              <h6>
                            REMAINING
              </h6>
</div>,
            <div
              className="divider"
>
              <h4>
                            SNAPS
              </h4>
              <span>
                            THIS WEEK
              </span>
</div>,
            <div
              className="received-points"
>
              <span
                            className="points"
              />
              <h6>
                            RECEIVED
              </h6>
</div>,
          ],
          "className": "current-points",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="points"
>
                  1000
</span>,
                <h6>
                  REMAINING
</h6>,
              ],
              "className": "remaining-points",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1000,
                  "className": "points",
                },
                "ref": null,
                "rendered": 1000,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "REMAINING",
                },
                "ref": null,
                "rendered": "REMAINING",
                "type": "h6",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  SNAPS
</h4>,
                <span>
                  THIS WEEK
</span>,
              ],
              "className": "divider",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SNAPS",
                },
                "ref": null,
                "rendered": "SNAPS",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "THIS WEEK",
                },
                "ref": null,
                "rendered": "THIS WEEK",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="points"
/>,
                <h6>
                  RECEIVED
</h6>,
              ],
              "className": "received-points",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "points",
                },
                "ref": null,
                "rendered": 0,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "RECEIVED",
                },
                "ref": null,
                "rendered": "RECEIVED",
                "type": "h6",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            AWARD
</h4>,
          "className": "award-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "AWARD",
          },
          "ref": null,
          "rendered": "AWARD",
          "type": "h4",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="award"
>
              <div
                            className="send-input"
              >
                            <h5
                                          className="send"
                            >
                                          Send
                            </h5>
                            <input
                                          name="points"
                                          onChange={[Function]}
                                          placeholder="QTY"
                                          type="text"
                                          value=""
                            />
              </div>
              <div
                            className="recipient-input"
              >
                            <h5
                                          className="snaps"
                            >
                                          snaps to:
                            </h5>
                            <input
                                          name="recipient"
                                          onChange={[Function]}
                                          onKeyDown={[Function]}
                                          placeholder="find a teammate"
                                          type="text"
                                          value=""
                            />
                            <ul>
                                          
                            </ul>
              </div>
</div>,
            <textarea
              className="note-input"
              name="note"
              onChange={[Function]}
              placeholder="add a message"
              type="text"
              value=""
/>,
            <button
              onClick={[Function]}
>
              SEND
</button>,
            <div
              className="transaction-message"
>
              <h3 />
</div>,
          ],
          "className": "points-block",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="send-input"
>
                  <h5
                                    className="send"
                  >
                                    Send
                  </h5>
                  <input
                                    name="points"
                                    onChange={[Function]}
                                    placeholder="QTY"
                                    type="text"
                                    value=""
                  />
</div>,
                <div
                  className="recipient-input"
>
                  <h5
                                    className="snaps"
                  >
                                    snaps to:
                  </h5>
                  <input
                                    name="recipient"
                                    onChange={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="find a teammate"
                                    type="text"
                                    value=""
                  />
                  <ul>
                                    
                  </ul>
</div>,
              ],
              "className": "award",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h5
                      className="send"
>
                      Send
</h5>,
                    <input
                      name="points"
                      onChange={[Function]}
                      placeholder="QTY"
                      type="text"
                      value=""
/>,
                  ],
                  "className": "send-input",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Send",
                      "className": "send",
                    },
                    "ref": null,
                    "rendered": "Send",
                    "type": "h5",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "points",
                      "onChange": [Function],
                      "placeholder": "QTY",
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h5
                      className="snaps"
>
                      snaps to:
</h5>,
                    <input
                      name="recipient"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="find a teammate"
                      type="text"
                      value=""
/>,
                    <ul>
                      
</ul>,
                  ],
                  "className": "recipient-input",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "snaps to:",
                      "className": "snaps",
                    },
                    "ref": null,
                    "rendered": "snaps to:",
                    "type": "h5",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "recipient",
                      "onChange": [Function],
                      "onKeyDown": [Function],
                      "placeholder": "find a teammate",
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [],
                    },
                    "ref": null,
                    "rendered": Array [],
                    "type": "ul",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "note-input",
              "name": "note",
              "onChange": [Function],
              "placeholder": "add a message",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SEND",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "SEND",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3 />,
              "className": "transaction-message",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="current-points"
>
            <div
                        className="remaining-points"
            >
                        <span
                                    className="points"
                        >
                                    1000
                        </span>
                        <h6>
                                    REMAINING
                        </h6>
            </div>
            <div
                        className="divider"
            >
                        <h4>
                                    SNAPS
                        </h4>
                        <span>
                                    THIS WEEK
                        </span>
            </div>
            <div
                        className="received-points"
            >
                        <span
                                    className="points"
                        />
                        <h6>
                                    RECEIVED
                        </h6>
            </div>
</div>,
          <div
            className="award-header"
>
            <h4>
                        AWARD
            </h4>
</div>,
          <div
            className="points-block"
>
            <div
                        className="award"
            >
                        <div
                                    className="send-input"
                        >
                                    <h5
                                                className="send"
                                    >
                                                Send
                                    </h5>
                                    <input
                                                name="points"
                                                onChange={[Function]}
                                                placeholder="QTY"
                                                type="text"
                                                value=""
                                    />
                        </div>
                        <div
                                    className="recipient-input"
                        >
                                    <h5
                                                className="snaps"
                                    >
                                                snaps to:
                                    </h5>
                                    <input
                                                name="recipient"
                                                onChange={[Function]}
                                                onKeyDown={[Function]}
                                                placeholder="find a teammate"
                                                type="text"
                                                value=""
                                    />
                                    <ul>
                                                
                                    </ul>
                        </div>
            </div>
            <textarea
                        className="note-input"
                        name="note"
                        onChange={[Function]}
                        placeholder="add a message"
                        type="text"
                        value=""
            />
            <button
                        onClick={[Function]}
            >
                        SEND
            </button>
            <div
                        className="transaction-message"
            >
                        <h3 />
            </div>
</div>,
        ],
        "className": "Transaction",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="remaining-points"
>
                <span
                                className="points"
                >
                                1000
                </span>
                <h6>
                                REMAINING
                </h6>
</div>,
              <div
                className="divider"
>
                <h4>
                                SNAPS
                </h4>
                <span>
                                THIS WEEK
                </span>
</div>,
              <div
                className="received-points"
>
                <span
                                className="points"
                />
                <h6>
                                RECEIVED
                </h6>
</div>,
            ],
            "className": "current-points",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="points"
>
                    1000
</span>,
                  <h6>
                    REMAINING
</h6>,
                ],
                "className": "remaining-points",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1000,
                    "className": "points",
                  },
                  "ref": null,
                  "rendered": 1000,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "REMAINING",
                  },
                  "ref": null,
                  "rendered": "REMAINING",
                  "type": "h6",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    SNAPS
</h4>,
                  <span>
                    THIS WEEK
</span>,
                ],
                "className": "divider",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SNAPS",
                  },
                  "ref": null,
                  "rendered": "SNAPS",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "THIS WEEK",
                  },
                  "ref": null,
                  "rendered": "THIS WEEK",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="points"
/>,
                  <h6>
                    RECEIVED
</h6>,
                ],
                "className": "received-points",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "className": "points",
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "RECEIVED",
                  },
                  "ref": null,
                  "rendered": "RECEIVED",
                  "type": "h6",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
              AWARD
</h4>,
            "className": "award-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "AWARD",
            },
            "ref": null,
            "rendered": "AWARD",
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="award"
>
                <div
                                className="send-input"
                >
                                <h5
                                                className="send"
                                >
                                                Send
                                </h5>
                                <input
                                                name="points"
                                                onChange={[Function]}
                                                placeholder="QTY"
                                                type="text"
                                                value=""
                                />
                </div>
                <div
                                className="recipient-input"
                >
                                <h5
                                                className="snaps"
                                >
                                                snaps to:
                                </h5>
                                <input
                                                name="recipient"
                                                onChange={[Function]}
                                                onKeyDown={[Function]}
                                                placeholder="find a teammate"
                                                type="text"
                                                value=""
                                />
                                <ul>
                                                
                                </ul>
                </div>
</div>,
              <textarea
                className="note-input"
                name="note"
                onChange={[Function]}
                placeholder="add a message"
                type="text"
                value=""
/>,
              <button
                onClick={[Function]}
>
                SEND
</button>,
              <div
                className="transaction-message"
>
                <h3 />
</div>,
            ],
            "className": "points-block",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="send-input"
>
                    <h5
                                        className="send"
                    >
                                        Send
                    </h5>
                    <input
                                        name="points"
                                        onChange={[Function]}
                                        placeholder="QTY"
                                        type="text"
                                        value=""
                    />
</div>,
                  <div
                    className="recipient-input"
>
                    <h5
                                        className="snaps"
                    >
                                        snaps to:
                    </h5>
                    <input
                                        name="recipient"
                                        onChange={[Function]}
                                        onKeyDown={[Function]}
                                        placeholder="find a teammate"
                                        type="text"
                                        value=""
                    />
                    <ul>
                                        
                    </ul>
</div>,
                ],
                "className": "award",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h5
                        className="send"
>
                        Send
</h5>,
                      <input
                        name="points"
                        onChange={[Function]}
                        placeholder="QTY"
                        type="text"
                        value=""
/>,
                    ],
                    "className": "send-input",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Send",
                        "className": "send",
                      },
                      "ref": null,
                      "rendered": "Send",
                      "type": "h5",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "name": "points",
                        "onChange": [Function],
                        "placeholder": "QTY",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h5
                        className="snaps"
>
                        snaps to:
</h5>,
                      <input
                        name="recipient"
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        placeholder="find a teammate"
                        type="text"
                        value=""
/>,
                      <ul>
                        
</ul>,
                    ],
                    "className": "recipient-input",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "snaps to:",
                        "className": "snaps",
                      },
                      "ref": null,
                      "rendered": "snaps to:",
                      "type": "h5",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "name": "recipient",
                        "onChange": [Function],
                        "onKeyDown": [Function],
                        "placeholder": "find a teammate",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [],
                      },
                      "ref": null,
                      "rendered": Array [],
                      "type": "ul",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "note-input",
                "name": "note",
                "onChange": [Function],
                "placeholder": "add a message",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SEND",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "SEND",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3 />,
                "className": "transaction-message",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
